eng/Versions.props

```diff
@@ -43,8 +43,8 @@
   </ItemGroup>
   <PropertyGroup>
     <!-- For source generator support we need to target a pinned version in order to be able to run on older versions of Roslyn -->
```

Comment:
Sort of. This PR isn't intended to merged as-is. Just a quick spike to see what it would look like to convert. We'll still need to figure out the multi-targeting approach if we do this.